src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1076,6 +1087,37 @@ public function getEnvCounters()
     }
 
     /**
+     * Replaces parameter placeholders (%name%) and unescapes percent signs.
+     *
+     * @param mixed $value            A value
+     * @param mixed $resolveEnvValues Whether %env(VAR)% parameters should be replaced by the value of the corresponding environment variable or not
+     * @param array $resolving        An array of keys that are being resolved (used internally to detect circular references)
+     *
+     * @return mixed The resolved value
+     */
+    public function resolveValue($value, $resolveEnvValues = false, array $resolving = array())
```

Comment:
This makes the resolution stateless. `ParameterBag::resolveValue` has the same strategy on this topic. I'd prefer keeping it this way.